### PR TITLE
MEPTS-51_TXPVLS_Changes: Added minor changes on OnArtForLessThanXmonthsCalcultion

### PR DESCRIPTION
**Pull request by Guimino**
getting viraloadConcept from HivMetadata  and  created method to check if objects are  not null

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/pvls/OnArtForLessThanXmonthsCalcultion.java [hunk 1]
```diff
@@ -1,31 +1,37 @@
...
 	public CalculationResultMap evaluate(Collection<Integer> cohort, Map<String, Object> params, PatientCalculationContext context) {
 		
 		CalculationResultMap map = new CalculationResultMap();
-		Concept viralLoadConceptuuid = Context.getConceptService().getConceptByUuid("e1d6247e-1d5f-11e0-b929-000c29ad1d07");
+		Concept viralLoadConcept = hivMetadata.getHivViralLoadConcept();
+		
 		// get data inicio TARV
 		CalculationResultMap arvsInitiationDateMap = calculate(new InitialArtStartDateCalculation(), cohort, context);
-		CalculationResultMap lastVl = EptsCalculations.lastObs(viralLoadConceptuuid, cohort, context);
+		CalculationResultMap lastVl = EptsCalculations.lastObs(viralLoadConcept, cohort, context);
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/pvls/OnArtForLessThanXmonthsCalcultion.java]: I always get a null pointer exception when I Autowire components and use them, this made me revert to the current way am using. Did you test that and make sure the NPE is NOT being thrown?

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxPvlsCohortQueries.java [hunk 2]
```diff
@@ -52,7 +51,7 @@
 	 */
 	public CohortDefinition getPatientsWhoAreLessThan3MonthsOnArt() {
 		CalculationCohortDefinition cd = new CalculationCohortDefinition("On ART for less than 3 months",
-		        new OnArtForLessThanXmonthsCalcultion());
+		        Context.getRegisteredComponents(OnArtForLessThanXmonthsCalcultion.class).get(0));
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxPvlsCohortQueries.java]: My evaluator will need new objects(instances) of the calculation being created, why would we want to go through the Context for this? any benefits for this that the previous method?

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/pvls/OnArtForLessThanXmonthsCalcultion.java (hunk 1 above)
Comment (Guimino) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/pvls/OnArtForLessThanXmonthsCalcultion.java]: @ningosi , that's weird, but there is no NPE in that code, you can check. maybe you were not putting the correct annotation in the component class.

Comment (nthfloor) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/pvls/OnArtForLessThanXmonthsCalcultion.java]: Works for me too, I am not getting a NPE. Agree with Guimino's comment about the annotations and we've used this approach in the other reports so should work here too.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxPvlsCohortQueries.java (hunk 2 above)
Comment (Guimino) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxPvlsCohortQueries.java]: I don't think we need to  be instantiating calculation classes to the evaluator, the context will provide one instance of the OnArtForLessThanXmonthsCalcultion

**Pull request merged**